### PR TITLE
Edit Security Guide's Session Guidelines & Custom Credentials [skip ci]

### PR DESCRIPTION
**Pull request by matildasmeds**
* Edit Session Guidelines to achieve tighter prose and accuracy
* Remove mentions related to earlier Rails versions
* Add links to ActionController guide and Custom Credentials part
* Clarify Custom Credentials part

This is a new attempt at [PR#28686](https://github.com/rails/rails/pull/28686), which I'm closing now. Sometimes it is easier to make a new one.

All the comments from the original PR have been taken into account, including the latest suggestions from @oniofchaos. :vulcan_salute:

### CODE REVIEW

**Comment by Guillerbr** [comment]
good!

File: guides/source/security.md [hunk 1]
```diff
@@ -32,27 +32,17 @@ In order to develop secure web applications you have to keep up to date on all l
 Sessions
 --------
 
-A good place to start looking at security is with sessions, which can be vulnerable to particular attacks.
+This chapter describes some particular attacks related to sessions, and security measures to protect your session data.
 
 ### What are Sessions?
 
-NOTE: _HTTP is a stateless protocol. Sessions make it stateful._
+INFO: Sessions enable the application to maintain user-specific state, while users interact with the application. Once the user has been authenticated, an active session enables the user to keep using the application, without requiring authentication on each request.
```
[the remark this answers is not in the document]
Comment (matildasmeds) [reply guides/source/security.md]: @derekprior I really like this one!

File: guides/source/security.md [hunk 2]
```diff
@@ -1235,6 +1193,11 @@ version:
 Rails.application.credentials.some_api_key! # => raises KeyError: :some_api_key is blank
 ```
 
+TIP: Learn more about credentials with `rails credentials:help`.
+
+WARNING: Keep master key safe. Do not commit master key.
```
[the remark this answers is not in the document]
Comment (matildasmeds) [reply guides/source/security.md]: @derekprior Makes sense, I prefer the first one, because it is shorter. :)

**Comment by matildasmeds** [comment]
r? @rafaelfranca (this PR is essentially the same as [PR28686](https://github.com/rails/rails/pull/28686)
Any further comments? From my perspective this PR is ready to be merged. :)

**Comment by matildasmeds** [comment]
Fixed new merge conflicts. Is there still something I could do, to get this PR merged?

**Pull request merged**